st2client/st2client/base.py

```diff
@@ -124,7 +124,7 @@ def get_client(self, args, debug=False):
         password = credentials.get('password', None)
         cache_token = rc_config.get('cli', {}).get('cache_token', False)
 
-        if credentials:
+        if username:
```

Comment:
Agreed - and we should start with tests and making sure we handle all the edge cases.